src/app/pages/applications/catalog/catalog.component.ts

```diff
@@ -143,11 +121,22 @@ export class CatalogComponent implements OnInit {
     this.catalogApps = [];
     this.isLoading = true;
     this.showLoadStatus(EmptyType.Loading);
+    this.catalogSyncJobs = [];
 
     this.appService.getAllCatalogItems().pipe(untilDestroyed(this)).subscribe((catalogs) => {
       this.noAvailableCatalog = true;
       for (let i = 0; i < catalogs.length; i++) {
         const catalog = catalogs[i];
+        if (!catalog.cached) {
+          if (catalog.caching_job) {
```

Comment:
Are there other conditions under `if (!catalog.cached)` in the future? If not, how about combining them into a single?